### PR TITLE
Updates for new typing.

### PR DESCRIPTION
**Pull request by tasseff**
Here is a first shot at conforming to the new typing convention in InfrastructureModels packages. Let me know if this looks okay. More updates (on a separate branch) to come for greater conformity with current version of PowerModels.

### CODE REVIEW

File: src/core/base.jl [hunk 1]
```diff
@@ -216,38 +216,20 @@ Some of the common keys include:
...
+        #        try
+        #            item_lookup = Dict{Int, Any}([(parse(Int, k), v) for (k, v) in item])
+        #            ref[Symbol(key)] = item_lookup
+        #        catch
+        #            ref[Symbol(key)] = item
+        #        end
+        #    else
+        #        ref[Symbol(key)] = item
+        #    end
+        #end
```
Comment (ccoffrin) [inline src/core/base.jl]: Tests pass so you can drop these commented lines.

File: src/core/data.jl [hunk 2]
```diff
@@ -1,3 +1,11 @@
+"Maps component types to status parameters."
+const wm_component_status = Dict("nodes" => "status", "pipes" => "status",
+                                 "pumps" => "status", "tanks" => "status")
```
Comment (ccoffrin) [inline src/core/data.jl]: You don't need this dict, becouse this code conforms to the `status` field standards.  PowerModels does not due to some legacy data model, but we are planning to resolve that in the next 6 months or so.

File: src/core/data.jl [hunk 3]
```diff
@@ -1,3 +1,11 @@
...
+
+"Maps component types to inactive status values."
+const wm_component_status_inactive = Dict("nodes" => 0, "pipes" => 0,
+                                          "pumps" => 0, "tanks" => 0)
```
Comment (ccoffrin) [inline src/core/data.jl]: Same remark as the `wm_component_status` comment.

File: src/core/solution.jl [hunk 4]
```diff
@@ -55,35 +55,44 @@ function get_solution(wm::GenericWaterModel, sol::Dict{String, <:Any})
...
 end
 
-function add_pipe_flow_rate_setpoint(sol, wm::GenericWaterModel)
+function solution_owf!(wm::AbstractWaterModel, sol::Dict{String,<:Any})
+    add_setpoint_pipe_flow!(sol, wm)
+    add_setpoint_pump_flow!(sol, wm)
+    #add_setpoint_pump_status!(sol, wm)
+    add_setpoint_pipe_resistance!(sol, wm)
+    add_setpoint_node_head!(sol, wm)
+    #add_setpoint_reservoir!(sol, wm)
```
Comment (ccoffrin) [inline src/core/solution.jl]: Drop commented lines?

File: src/core/solution.jl [hunk 5]
```diff
@@ -119,8 +128,250 @@ function add_pipe_resistance_setpoint(sol, wm::GenericWaterModel)
...
+            a = link["index"]
+            sol_item = sol_dict[i] = get(sol_dict, i, Dict{String, Any}())
+            x_res, r_id = findmin(ref(wm, :resistance, a))
+            sol_item["r"] = ref(wm, :resistance, a)[r_id]
+        end
+    end
+end
+
+"adds values based on JuMP variables"
+function add_setpoint!(
```
Comment (ccoffrin) [inline src/core/solution.jl]: Looking at this reminds me that we should setup some abstract solution building tools in InfrastructureModels, i.e. https://github.com/lanl-ansi/InfrastructureModels.jl/issues/1.  Feel free to remind me about next month.  :-)

File: test/runtests.jl [hunk 6]
```diff
@@ -2,30 +2,26 @@ using WaterModels
 import InfrastructureModels
 import Memento
 
+# Suppress warnings during testing.
+Memento.setlevel!(Memento.getlogger(InfrastructureModels), "error")
```
Comment (ccoffrin) [inline test/runtests.jl]: Looking at this makes me think we should probably add the `logger_config!` function to InfrastructureModels.

File: test/runtests.jl [hunk 7]
```diff
@@ -2,30 +2,26 @@ using WaterModels
...
+WaterModels.logger_config!("error")
+
+# For checking status codes
+import MathOptInterface
+const MOI = MathOptInterface
```
Comment (ccoffrin) [inline test/runtests.jl]: You should be able to drop these in the tests if you prefer becouse WaterModels exports the MOI status codes.  This is a style choice each package can make, i.e. If the tests should emulate a user calling `using` or a package developer calling `import`.

**Comment by tasseff** [comment]
Thanks for the comments! I've addressed your comments and think it's in better shape, now. Unless there are any objections, I'm going to merge this with master.

**Comment by ccoffrin** [comment]
Sounds good to me.  PowerModels will be merging the type update to master early next week, but we are holding off tagging a new release until the end of Sept.

**Pull request merged**